build/package/admin.Dockerfile

```diff
@@ -0,0 +1,69 @@
+# Base Go environment
+# -------------------
+FROM golang:1.21-alpine as base
+WORKDIR /hatchet
+
+RUN apk update && apk add --no-cache gcc musl-dev git protoc protobuf-dev
+
+COPY go.mod go.sum ./
+COPY /api ./api
+COPY /api-contracts ./api-contracts
+COPY /internal ./internal
+COPY /pkg ./pkg
+COPY /hack ./hack
+COPY /prisma ./prisma
+COPY /cmd ./cmd
+
+RUN go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28
+RUN go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2
+RUN go install github.com/deepmap/oapi-codegen/v2/cmd/oapi-codegen@latest
```

Comment:
1. move these to before any COPY
2. use fixed versions to prevent any version issues, e.g. `github.com/deepmap/oapi-codegen/v2/cmd/oapi-codegen@v2.0.0`